### PR TITLE
Document optional path parameters

### LINKED ISSUE #339: Document optional path param routing
https://luckyframework.org/guides/http-and-routing/routing-and-params

https://github.com/luckyframework/lucky_router/pull/18

### PR DESCRIPTION
**Pull request by stephendolan**
This PR aims to close #339

This section ended up being pretty minimal, but it exposes the API and (I think) provides a nice example where you may want to use it, since the purpose escaped me initially.

### CODE REVIEW

**Review by paulcsmith — APPROVED** [review]
This is an excellent example. Thanks!

File: src/actions/guides/http_and_routing/routing_and_params.cr [hunk 1]
```diff
@@ -117,6 +117,26 @@ class Guides::HttpAndRouting::RoutingAndParams < GuideAction
...
+        if day
+          plain_text "I'll show all posts on a specific day!"
+        else
+          plain_text "I'll show all posts in a given month!"
+        end
+      end
+    end
+    ```
+
+    In the above example, we require that the `Post` index route has both a `:year` and `:day` provided, but allow users to optionally route to a specific `:day` as well.
```
Comment (stephendolan) [inline src/actions/guides/http_and_routing/routing_and_params.cr:138]: Darn it, @paulcsmith, I just realized that this should be "`:year` and `:month`" 😭 

Should I open a PR to correct or is there a way to change that first `:day` to `:month` in master from your end?

Comment (paulcsmith) [reply src/actions/guides/http_and_routing/routing_and_params.cr:138]: I can change it pretty easily on master :D

Comment (stephendolan) [reply src/actions/guides/http_and_routing/routing_and_params.cr:138]: Thanks man, sorry about that! I think it was off just enough that my brain filtered it out 😛

**Pull request merged**